src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java

```diff
@@ -0,0 +1,240 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.configuration.Configurations;
+import com.minecolonies.coremod.util.ServerUtils;
+import net.minecraft.block.Block;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Blocks;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+import static com.minecolonies.coremod.commands.AbstractSingleCommand.Commands.COLONYTP;
+
+/**
+ * this command is made to TP a player to a safe random spot that is not to close to another colony
+ * Need to add a configs permissions check
+ * Need to allow OPs to send players ./mc ctp (Player) if player is not allowed
+ */
+public class ColonyTPCommand extends AbstractSingleCommand
+{
+    public static final  String DESC = "ctp";
+    private static final int ATTEMPTS = 4;
+    private static final int UPPER_BOUNDS = Configurations.maxDistanceFromWorldSpawn * 2;
```

Comment:
UPPER is written that way as the rand wont do negative numbers.  So this would generate a rand from 0 to 16k  then I subtract the 8k they give and can get the negative there.  

if rnd = 0-8k i will get a negative from that (rnd - 8000)
if rnd = 8k-16k i will get a positive

Hope that makes sense